internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -94,6 +97,29 @@ func (r *BundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ c
 	}()
 	u.UpdateStatus(updater.EnsureObservedGeneration(bundle.Generation))
 
+	res, err := r.Finalizers.Finalize(ctx, bundle)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
Do we need to surface this error on the Bundle's status or is persisting that error state a leaky abstraction?

Reply:
Oh good catch. Yeah I think we need to surface this error. My only hesitation is that I have in my head that you can't update an object in certain ways once it has a deletion timestamp. However, I'm guessing that status updates must be allowed. Otherwise how would you ever know why a finalizer was failing to be removed?

Reply:
> In short, what’s happened is that the object was updated, not deleted. That’s because Kubernetes saw that the object contained finalizers and put it into a read-only state. The deletion timestamp signals that the object can only be read, with the exception of removing the finalizer key updates. In other words, the deletion will not be complete until we edit the object and remove the finalizer.

I'm not entirely reading through [Using Finalizers to Control Deletion](https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/) whether updating a sub-resource once the finalizer is in place would be a valid workflow? Maybe this is something we can spec out locally using a simple main.go?

Reply:
I just did a manual test using the kubectl edit-status plugin. Status updates are allowed.

I'll update this to set a condition relating to this error.